pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3420,17 +3448,29 @@ public void readEntryFailed(ManagedLedgerException exception, Object ctx) {
     public synchronized void triggerCompaction()
             throws PulsarServerException, AlreadyRunningException {
         if (currentCompaction.isDone()) {
+            if (!lock.readLock().tryLock()) {
+                log.info("[{}] Topic is closing or deleting, skip triggering compaction -lock", topic);
+                return;
+            }
+            try {
+                if (isClosingOrDeleting) {
+                    log.info("[{}] Topic is closing or deleting, skip triggering compaction -flag", topic);
```

Comment:
Should them be warning logs?  BTW, the `-lock` and `-flag` words in the logs are not readable.

Reply:
I think this should be an expected behavior. It should skip triggering compaction if topic is closing or deleting.
I optimized the logs.